partiql-eval/src/main/kotlin/org/partiql/eval/internal/Parameters.kt

```diff
@@ -0,0 +1,26 @@
+package org.partiql.eval.internal
+
+import org.partiql.spi.value.Datum
+
+/**
+ * Statement execution parameters.
+ */
+internal class Parameters(private val values: Array<Datum>) {
```

Comment:
Mentioned in other comment, but I don't believe this is necessary.

Reply:
Ack, I can remove this, but just letting you know it's not coming from JDBC.

You can always plan a "parameterized" node as dynamic which would allow repeated execution of prepared statements with different parameters. This should seriously cut down on a lot of unnecessary work, but we could even do better.

You also don't need any info about the parameter other than the type, so you can always plan with parameters which are only defined by their types. Then you can cache prepared plans and invoke with the differing parameters. Recall that the ability to do so was a direct ask from customers who wanted to plan once and repeatedly execute while changing some parameter values.

You can think of the prepared statement as producing a function that can be invoked. As-is, you are unable to provide any inputs to that function. In other words, our query compiles to `fun query(): Datum` but parameters allows us to compile the query to `fun query(p1: T1, p2: T2, ...): Datum`.

Reply:
The direct ask was resolved using our plugin model -- and the executable maintained a reference to a table (which was mutable by design). This might just be a preemptive optimization, since we don't allow for parameters in the plan itself (yet?).

Reply:
I see, use a mutable table rather than an execution parameter. That works for the tools available, and doesn't need any other changes and achieves the same thing but I would push back on compiling over a mutable value. I understand it solved the problem with what is available, but that is like passing mutable pointers to a function call and replacing the data at the pointer before invoking again, rather than just passing the new state. 

I can remove this and may tinker with helpers in partiql-lang but the grammar and AST have parameter expressions. Also this could be added later. Parameters are just missing from v1 planner/plans afaik